src/tensorflow/tensorflow-rnnlm-lib.h

```diff
@@ -0,0 +1,156 @@
+  std::string eos_symbol;
+  int32 num_jobs;  // 0 means unlimited
+
+  KaldiTfRnnlmWrapperOpts() : unk_symbol("<oos>"), eos_symbol("</s>"),
+                              num_jobs(1) {}
+
+  void Register(OptionsItf *opts) {
+    opts->Register("unk-symbol", &unk_symbol, "Symbol for out-of-vocabulary "
+                   "words in rnnlm.");
+    opts->Register("eos-symbol", &eos_symbol, "End of setence symbol in "
+                   "rnnlm.");
+    opts->Register("num-jobs", &num_jobs, "Number of jobs for TF computation; "
+                   "0 means unlimited. (default = 1)");
+  }
+};
+
+class KaldiTfRnnlmWrapper {
+ public:
+  KaldiTfRnnlmWrapper(const KaldiTfRnnlmWrapperOpts &opts,
+                      const std::string &rnn_wordlist,
+                      const std::string &word_symbol_table_rxfilename,
+                      const std::string &unk_prob_file,
+                      const std::string &tf_model_path);
+
+  ~KaldiTfRnnlmWrapper() {
+    session_->Close();
+  }
+
+  int32 GetEos() const { return eos_; }
+
+  // get an all-zero Tensor of the size that matches the hidden state of the TF model
+  const Tensor& GetInitialContext() const;
+
+  // get the 2nd-to-last layer of RNN when feeding input of
+  // (initial-context, sentence-boundary)
+  const Tensor& GetInitialCell() const;
+
+  // compute p(word | wseq) and return the log of that
+  // the computation used the input cell,
+  // which is the 2nd-to-last layer of the RNNLM associated with history wseq;
+  //
+  // and we generate (context_out, new_cell) by passing (context_in, word) into the model
+  // if the last 2 pointers are NULL we don't query then in TF session
+  BaseFloat GetLogProb(int32 word, // need the FST word label for computing OOS cost
+                       int32 fst_word,
+                       const Tensor &context_in,  // context to pass into RNN
+                       const Tensor &cell_in,     // 2nd-to-last layer
+                       Tensor *context_out,
+                       Tensor *new_cell);
+
+  // since usually we have a smaller vocab in RNN than the whole vocab,
+  // we use this mapping during rescoring
+  std::vector<int> fst_label_to_rnn_label_;
+  std::vector<std::string> rnn_label_to_word_;
+  std::vector<std::string> fst_label_to_word_;
+ private:
+  void ReadTfModel(const std::string &tf_model_path, int32 num_jobs);
+
+  // do queries on the session to get the initial tensors (cell + context)
+  void AcquireInitialTensors();
+
+  KaldiTfRnnlmWrapperOpts opts_;
+  Tensor initial_context_;
+  Tensor initial_cell_;
+
+  // this corresponds to the FST symbol table
+  int32 num_total_words;
+  // this corresponds to the RNNLM symbol table
+  int32 num_rnn_words;
+
+  Session* session_;  // for TF computation; pointer owned here
+  int32 eos_;
+  int32 oos_;
+
+  std::vector<float> unk_costs_;  // extra cost for OOS symbol in RNNLM
+
+  KALDI_DISALLOW_COPY_AND_ASSIGN(KaldiTfRnnlmWrapper);
+};
+
+class TfRnnlmDeterministicFst
```

Comment:
put the colon directly after TfRnnlmDeterministicFst, like in  other Kaldi code.